resources/config_external_image_storage.rb

```diff
@@ -0,0 +1,61 @@
+#
+# Cookbook Name:: grafana
+# Resource:: config_external_image_storage
+#
+# Copyright 2018, Sous Chefs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Configures the installed grafana instance
+
+property  :instance_name,           String,         name_property: true
+property  :conf_directory,          String,         default: '/etc/grafana'
+property  :config_file,             String,         default: lazy { ::File.join(conf_directory, 'grafana.ini') }
+property  :storage_provider,        String,         required: true, default: 's3'
```

Comment:
If we are supporting multiple providers here then we need to look into naming of variables to work out which are required for which provider, it may be better to create as a specific s3 resource?

Reply:
@Xorima you are correct. Now I have renamed the resource to be S3 specific.